### DISCUSSION TITLE
".rotation" should refer to rotation degrees

### DISCUSSION THREAD

**Discussion by Starkandco** [godotengine/godot-proposals#8491] [Scripting]
I have seen a few people and myself bash our heads off trying to update rotation in code. I'm assuming from someone else's observation that .rotation refers to the transform of rotation by a number of pixels. If that was the case I would be suggesting that a rotation_pixels property is created to allow the rotation property to be used to reference the rotation degrees property, which is much more intuitive than pixels/what the actual rotation property is if I have that wrong?

**Comment by AThousandShips** [comment]
It is rotation in radians, as the [documentation says](https://docs.godotengine.org/en/stable/classes/class_node2d.html#class-node2d-property-rotation), please read the documentation if you are unsure about what something does, or just read the tooltip

**Reply by Starkandco** [reply]
Sure, thanks for clarifying. I think then a rotation_radians would be more intuitive? Why is the rotation degrees used in the inspector for example, I think most people think rotation in terms of degrees. Also this is inconsistent in user experience

**Reply by AThousandShips** [reply]
It isn't used in the editor, it's converted

The default property is in radians because it is the most commonly used in programming, changing this would break compatibility so can't be done until 5.0

There's no inconsistency here

> I think most people think rotation in terms of degrees

You are unfortunately wrong, in regards to doing actual maths with rotations radians is the most common, people might *think* in degrees, but most do the actual calculations in radians because it's better and aligns with the code methods

**Reply by Starkandco** [reply]
I think you're being unfair - There may be a valid reason for this, and I am not demanding change. But a user interacts with the inspector first, it is a "rotation" property that acts in degrees. Then they go to code and type .rotation and the property is different. Not everyone is coming at coding from a pure maths and knowledgeable perspective. Two people in the last 12 hours brought this question into a discord channel, only knowing that they wanted degrees was why I was able to help

radians could also still be used in code, just as .rotation_radians

**Reply by AThousandShips** [reply]
How am I being unfair? There's a `rotation_degrees` right there you can use, renaming it can't happen for years and there's very little reason to add confusion to users who are used to this

Also people being confused about this should just read the documentation, then they wouldn't be confused, them not doing that doesn't mean the code is confusing

**Reply by Starkandco** [reply]
You are saying there's no inconsistency here for example? I pointed to it just there. The inspector "Rotation" inconsistent with .rotation. And suggesting to read the documentation is all well and good but sometimes that doesn't happen and in that case here, if .rotation was in degrees and there was a separate rotation_radians for use by the engine and users alike, there wouldn't be any confusion. This design invites confusion as it stands. 

An easier approach than updating those would be to rename the inspector label for rotation to rotation degrees seeing this is what it is working with from a user perspective

**Reply by AThousandShips** [reply]
We could rename it to `rotation_radians`, it would confuse people, and couldn't be done until we release 5.0

I don't agree that there is an inconsistency, we can disagree on that, but I'm not being unfair for having an opinion

The fact that you get `rotation_degrees` as a suggestion when typing out rotation should help you notice it's not in degrees, and people not reading the documentation is not a reason to change things

The property is not going to be changed to be degrees, it would be much to big a change, and adding `radians` to it would be extra work as well, but that could be done, but then that is the change that should be proposed, but do understand that regardless of how strongly supported it won't happen until 5.0 (in several years most likely)

I'll leave the discussion here, but consider these points 🙂 

Have a nice day

**Reply by Starkandco** [reply]
You too.

**Comment by Starkandco** [comment]
I understand the implication of not changing something so central to the core (at least outside of a major release). Would it be possible instead to update the label shown in the inspector to "Rotation degrees" as I feel this would direct the user to use this property when they are first sitting down approaching a script to modify rotation

**Reply by AThousandShips** [reply]
Not as far as I know, the name is automatic from the property

**Reply by Starkandco** [reply]
Sure I guess then this proposal can be considered with all the potential issues involved taken into account for some major release or no. Thanks a lot for your time and sorry if I was harsh myself, I felt a bit dismissed but I don't want to act out of order.

**Reply by AThousandShips** [reply]
Understandable, I mainly tried to clarify the fact that this can't change in the short term, and the fact that radians is the preferred type for programming as most maths around it is built on it, using degrees would involve a lot of conversion all the time as all the underlying maths use radians, like `sine/cosine` and such

I'd suggest to make the proposal about renaming `rotation` to `rotation_radians` as it's the proposal that has any chance of happening IMO, changing the meaning to a different unit would break a lot of people's code and would be unpopular among many

But keep in mind the scale and impact 🙂, happy I could provide some context

**Reply by Starkandco** [reply]
Thanks a lot for the advice

**Reply by Starkandco** [reply]
I was thinking - if you changed the rotation property name, then the inspector name would show the rotation radian property with a degrees slider which I think is not intuitive either. Would that be right? I might close this and keep thinking about it if so, thanks again

Could we expose the current degrees property to the inspector instead of the radians one with a conversion?

**Reply by AThousandShips** [reply]
That would be the case yes, didn't even think about that, chances are though that by 5.0 changes could be made to how the system in question works
